apps/roam/src/components/settings/AsyncQuerySettings.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from "react";
+import { Switch, FormGroup, Classes } from "@blueprintjs/core";
+import { OnloadArgs } from "roamjs-components/types";
+
+export function AsyncQuerySettings({
+  extensionApi,
+}: {
+  extensionApi: OnloadArgs["extensionAPI"];
+}) {
+  const [isEnabled, setIsEnabled] = useState(false);
+
+  // Load initial state from settings
+  useEffect(() => {
+    try {
+      const savedSetting = extensionApi.settings.get("async-q");
+      setIsEnabled(!!savedSetting);
+    } catch (error) {
+      console.error("Failed to load async query setting:", error);
+      setIsEnabled(false);
+    }
+  }, []);
+
+  const handleToggle = async (event: React.FormEvent<HTMLInputElement>) => {
+    const newValue = event.currentTarget.checked;
+    // Update state first
+    setIsEnabled(newValue);
+    try {
+      extensionApi.settings.set("async-q", newValue);
+    } catch (error) {
+      console.error("Failed to save async query setting:", error);
+      setIsEnabled(!newValue);
+    }
+  };
+
+  return (
+    <div className="p-4">
+      <FormGroup
+        helperText="This will use Roam's Backend Query. It helps prevent the UI from freezing during large queries but is still in beta and may occasionally produce inaccurate results."
```

Comment:
Let's use the tooltip pattern that the rest of the settings currently use for additional information.

![image](https://github.com/user-attachments/assets/8b6b71c5-a784-4254-9d29-1333cf6c35cc)